opensrp-reveal/src/main/java/org/smartregister/reveal/presenter/ListTaskPresenter.java

```diff
@@ -581,10 +607,19 @@ public void onAddStructureClicked() {
 
     @Override
     public void onLocationValidated() {
-        if (cardDetails == null || !changeSprayStatus) {
-            startSprayForm(selectedFeature);
-        } else {
-            startSprayForm(selectedFeature, cardDetails.getPropertyType(), cardDetails.getSprayStatus(), cardDetails.getFamilyHead());
+        if (IRS.equals(selectedFeatureInterventionType)) {
+            if (cardDetails == null || !changeSprayStatus) {
+                startForm(selectedFeature, null, SPRAY_EVENT);
+            } else {
+                startForm(selectedFeature, cardDetails, SPRAY_EVENT);
+            }
+        } else if (MOSQUITO_COLLECTION.equals(selectedFeatureInterventionType)) {
+            // TODO: add global variable to change mosquito collection details
```

Comment:
can this be implemented

Reply:
This will be implemented in the PR to [this](https://github.com/OpenSRP/opensrp-client-reveal/issues/116) issue.